packages/manager/apps/telecom/src/app/telecom/pack/access-list/xdsl-access-list.controller.js

```diff
@@ -0,0 +1,121 @@
+import {
+  ELIGIBILITY,
+  PAGE_SIZE,
+  ACCESS_TYPE,
+} from './xdsl-access-list.constant';
+
+export default class XdslAccessListCtrl {
+  /* @ngInject */
+  constructor(coreURLBuilder, XdslAccessListService, TucToastError) {
+    this.coreURLBuilder = coreURLBuilder;
+    this.XdslAccessListService = XdslAccessListService;
+    this.TucToastError = TucToastError;
+  }
+
+  $onInit() {
+    this.ELIGIBILITY = ELIGIBILITY;
+    this.pageSize = PAGE_SIZE;
+    this.ACCESS_TYPE = ACCESS_TYPE;
+
+    this.links = {
+      copperClosureMore:
+        'https://www.ovhcloud.com/fr/lp/fermeture-reseau-cuivre/',
+    };
+
+    this.getServices();
+  }
+
+  getServices() {
+    this.services = [];
+    this.XdslAccessListService.getServices()
+      .then(({ data }) => {
+        const services = data;
+        this.buildServicesList(services);
+      })
+      .catch(({ data }) => {
+        this.TucToastError(data.message);
+      });
+  }
+
+  buildServicesList(services) {
+    // Retrieve service info for each service
+    services.forEach((serviceName) => {
+      this.XdslAccessListService.getServiceDetail(serviceName).then(
+        ({ data }) => {
+          const service = data;
+          if (service.accessType !== 'ftth') {
+            this.getListFiberEligibilities(serviceName, service);
+          } else {
+            this.services.push({
+              accessName: service.accessName,
+              accessType: service.accessType,
+              description: service.description,
+              packName: service.packName,
+              migrationAvailable: this.ELIGIBILITY.not_concerned,
+              accessLink: this.coreURLBuilder.buildURL(
+                'telecom',
+                `#/pack/${service.packName}/xdsl/${service.accessName}`,
+              ),
+              migrationLink: this.coreURLBuilder.buildURL(
+                'telecom',
+                `#/pack/${service.packName}/migration`,
+              ),
+            });
+          }
+        },
+      );
+    });
+  }
+
+  getListFiberEligibilities(serviceName, service) {
+    this.XdslAccessListService.getFiberEligibilityList(serviceName).then(
+      ({ data }) => {
+        const fiberEligibilities = data;
+        if (fiberEligibilities.length === 1) {
+          this.getFiberEligibility(serviceName, service, fiberEligibilities);
+        } else {
+          this.services.push({
```

Comment:
I see similar bunch of codes between controllers, maybe share somewhere the logic inside ?

Reply:
In future feature, we would refactor code between move and migration controllers, but for the moment, it's not the case so I need to update the 2 controllers.

Reply:
Yes it's similar but it's defined into 2 different places.